src/index.ts

```diff
@@ -605,6 +603,93 @@ export async function precomputedInit(
   return EppoPrecomputedJSClient.instance;
 }
 
+/**
+ * Initializes the Eppo precomputed client with configuration parameters.
+ *
+ * The purpose is for use-cases where the precomputed assignments are available from an external process
+ * that can bootstrap the SDK.
+ *
+ * This method should be called once on application startup.
+ *
+ * @param config - client configuration
+ * @returns a singleton precomputed client instance
+ * @public
+ */
+export interface IPrecomputedClientConfigSync {
+  precomputedConfigurationWire: string;
+  assignmentLogger?: IAssignmentLogger;
+  throwOnFailedInitialization?: boolean;
+}
+
+// TODO: remove when this interface is exported from the common library
+export interface IConfigurationWire {
+  version: number;
+  precomputed: {
+    subjectKey: string;
+    subjectAttributes: Record<string, AttributeType>;
+    fetchedAt: string;
+    response: {
+      createdAt: string;
+      format: FormatEnum;
+      obfuscated: boolean;
+      environment: Environment;
+      flags: Record<string, PrecomputedFlag>;
+    };
+  };
+}
+
+/**
+ * Initializes the Eppo precomputed client with configuration parameters.
+ *
+ * The purpose is for use-cases where the precomputed assignments are available from an external process
+ * that can bootstrap the SDK.
+ *
+ * This method should be called once on application startup.
+ *
+ * @param config - precomputed client configuration
+ * @returns a singleton precomputed client instance
+ * @public
+ */
+export function offlinePrecomputedInit(
+  config: IPrecomputedClientConfigSync,
+): EppoPrecomputedClient {
+  const throwOnFailedInitialization = config.throwOnFailedInitialization ?? true;
+
+  const configurationWire: IConfigurationWire = JSON.parse(config.precomputedConfigurationWire);
+  const { subjectKey, subjectAttributes, response } = configurationWire.precomputed;
+  const parsedResponse = response; // TODO: use a JSON.parse when the obfuscated version is usable
+
+  try {
+    const memoryOnlyPrecomputedStore = precomputedFlagsStorageFactory();
+    memoryOnlyPrecomputedStore
+      .setEntries(parsedResponse.flags)
+      .catch((err) =>
+        applicationLogger.warn('Error setting precomputed assignments for memory-only store', err),
+      );
+
+    EppoPrecomputedJSClient.instance.setSubjectAndPrecomputedFlagStore(
+      subjectKey,
+      subjectAttributes,
+      memoryOnlyPrecomputedStore,
+    );
+
+    if (config.assignmentLogger) {
+      EppoPrecomputedJSClient.instance.setAssignmentLogger(config.assignmentLogger);
+    }
+  } catch (error) {
+    applicationLogger.warn(
+      'Eppo SDK encountered an error initializing precomputed client, assignment calls will return the default value and not be logged',
+    );
+    if (throwOnFailedInitialization) {
+      throw error;
+    }
+  }
+
+  EppoPrecomputedJSClient.instance.setIsObfuscated(parsedResponse.obfuscated);
```

Comment:
`parsedResponse` also has the salt needed to hash flagKeys in `getFlag`

Reply:
Yup using the salt now